alicloud/resource_alicloud_cs_kubernetes_autoscaler.go

```diff
@@ -0,0 +1,552 @@
+	}
+
+	return DeleteAutoscaler(kubeConfPath)
+}
+
+// update scaling group config
+func UpdateScalingGroupConfiguration(client *connectivity.AliyunClient, groupId, userData string, labels string, taints string) (err error) {
+
+	describeScalingConfigurationsRequest := ess.CreateDescribeScalingConfigurationsRequest()
+	describeScalingConfigurationsResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		describeScalingConfigurationsRequest.RegionId = client.RegionId
+		describeScalingConfigurationsRequest.ScalingGroupId = groupId
+		return essClient.DescribeScalingConfigurations(describeScalingConfigurationsRequest)
+	})
+
+	configurations, ok := describeScalingConfigurationsResponse.(*ess.DescribeScalingConfigurationsResponse)
+
+	if ok != true {
+		return WrapError(fmt.Errorf("failed to parse DescribeScalingConfigurationsResponse of %s", groupId))
+	}
+
+	if err != nil {
+		return WrapErrorf(err, "failed to invoke CreateDescribeScalingConfigurations,because of %v", err)
+	}
+
+	addDebug("DescribeScalingConfigurations", configurations, describeScalingConfigurationsRequest)
+
+	if len(configurations.ScalingConfigurations.ScalingConfiguration) == 0 {
+		//todo  create configuration
+		return WrapError(fmt.Errorf("please create the default scaling configuration of group %s", groupId))
+	} else {
+		defaultConfiguration := configurations.ScalingConfigurations.ScalingConfiguration[0]
+		// modify the default one
+		modifyScalingConfigurationRequest := ess.CreateModifyScalingConfigurationRequest()
+		modifyScalingConfigurationResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			modifyScalingConfigurationRequest.RegionId = client.RegionId
+			modifyScalingConfigurationRequest.UserData = userData
+			modifyScalingConfigurationRequest.Tags = createScalingGroupTags(labels, taints)
+			modifyScalingConfigurationRequest.ScalingConfigurationId = defaultConfiguration.ScalingConfigurationId
+			return essClient.ModifyScalingConfiguration(modifyScalingConfigurationRequest)
+		})
+
+		if err != nil {
+			return WrapErrorf(err, "failed to invoke ModifyScalingConfiguration,because of %v", err)
+		}
+
+		addDebug("ModifyScalingConfiguration", modifyScalingConfigurationResponse, modifyScalingConfigurationRequest)
+	}
+	return nil
+}
+
+func GetScalingGroupSizeRange(client *connectivity.AliyunClient, groupId string) (min, max int, err error) {
+	describeScalingGroupRequest := ess.CreateDescribeScalingGroupsRequest()
+	describeScalingGroupRequest.RegionId = client.RegionId
+	describeScalingGroupRequest.ScalingGroupId1 = groupId
+
+	describeScalingGroupResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		return essClient.DescribeScalingGroups(describeScalingGroupRequest)
+	})
+
+	if err != nil {
+		return 0, 0, WrapErrorf(err, "failed to invoke DescribeScalingGroups,because of %v", err)
```

Comment:
Please change line 329 to `return 0, 0, WrapErrorf(err, DefaultErrorMsg, d.Id(), describeScalingGroupRequest.GetActionName(), AlibabaCloudSdkGoERROR) )`